scripts/zkai_d128_recursive_pcd_route_selector_gate.py

```
@@ -0,0 +1,910 @@
+    return json.dumps(value, sort_keys=True, separators=(",", ":"), ensure_ascii=False).encode("utf-8")
+
+
+def sha256_hex_bytes(data: bytes) -> str:
+    return hashlib.sha256(data).hexdigest()
+
+
+def sha256_hex_json(value: Any) -> str:
+    return sha256_hex_bytes(canonical_json_bytes(value))
+
+
+def read_json_file(path: pathlib.Path) -> Any:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `read_json_file`

(ANN401)